lib/connect/tls.js

```diff
@@ -20,7 +20,8 @@ function buildBuilder (mqttClient, opts) {
 
   function handleTLSerrors (err) {
     // How can I get verify this error is a tls error?
-    if (opts.rejectUnauthorized) {
+    // Only throw error if it is not an connection refused error to reconnect
+    if (opts.rejectUnauthorized && err && err.code !== 'ECONNREFUSED') {
```

Comment:
it would be the cleaner approach i think, but i never managed to get one, so i wrote it this way.
i also tried to write a test for it, but failed due to the fact that i can not kill the mocked secure server during a test